crates/ruma-client-api/src/error/kind_serde.rs

```diff
@@ -130,7 +130,9 @@ impl<'de> Visitor<'de> for ErrorKindVisitor {
 
         while let Some(key) = map.next_key()? {
             match key {
-                Field::ErrCode => set_field!(errcode),
+                Field::ErrCode => {
+                    set_field!(errcode);
```

Comment:
Why is there a semicolon here now, but not in the `set_field!` invocations below?

Reply:
This was from the [`semicolon-in-expressions-from-macros`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#semicolon-in-expressions-from-macros) although I don't fully get it. Something about not using semicolons until the macro is fully expanded?

Reply:
Right, I think I have an idea on how this was happening only here. Can you try removing this semicolon again, and also removing the one in line 103? Maybe that silences the lint.